airflow/callbacks/callback_requests.py

```diff
@@ -55,7 +53,7 @@ class TaskCallbackRequest(CallbackRequest):
     def __init__(
         self,
         full_filepath: str,
-        simple_task_instance: SimpleTaskInstance,
+        simple_task_instance,  #: SimpleTaskInstance, Avoid circular import
```

Comment:
unfortunatelly this causes circular dependency

We could either remove the typing there or move the `callback_request.py` to subfolder `airflow/callbacks/model/callback_requests.py`

WDYT?